service-proxy.js

```diff
@@ -77,6 +80,10 @@ function ServiceDispatchHandler(options) {
     });
     self.rateLimiterEnabled = options.rateLimiterEnabled;
 
+    self.partialAffinityEnabled = options.partialAffinityEnabled;
+    self.minPeersPerWorker = options.minPeersPerWorker || DEFAULT_MIN_PEERS_PER_WORKER;
+    self.minPeersPerRelay = options.minPeersPerRelay || DEFAULT_MIN_PEERS_PER_RELAY;
+
     self.egressNodes.on('membershipChanged', onMembershipChanged);
```

Comment:
Debatable whether that is useful. The affinity gets updated upon every advertise and these happen with some regularity. Doing the refresh on membership changes would only make the shift in traffic more sudden.